### ISSUE TITLE
New formulas for zeta(3) and other constants

### ISSUE THREAD

**Issue by fredrik-johansson** [flintlib/flint#1622]
On

http://www.numberworld.org/y-cruncher/news/2023.html#2023_10_25

are some new formulas that could speed up zeta(3), Catalan, gamma(1/3).

**Comment by fredrik-johansson** [comment]
Catalan's constant and zeta(3) are done in #1624. I'll leave this open in case anyone wants to do the state-of-the-art formulas for gamma(1/3), zeta(5), ...

BTW, we define the hypergeometric terms slightly differently on https://flintlib.org/doc/hypgeom.html than in the y-cruncher definition files (https://github.com/Mysticial/y-cruncher-Formulas/tree/master/Official%20Formulas). Basically, we index from 0 instead of 1, and the product starts at the second term rather than the first. This means that one has to shift the polynomials a bit. It looks like the y-cruncher definition leads to slightly simpler formulas in many cases, so we should perhaps switch to that definition, or have a flag for the offsets.

**Comment by fredrik-johansson** [comment]
Note: for log(2), the formula given in a comment in https://math.stackexchange.com/questions/4793437/are-these-fast-convergent-series-for-the-lemniscate-constant-and-gamma-frac1?noredirect=1#comment10236711_4793437 i.e.

```
void
log2(arb_t s, slong prec)
{
    hypgeom_t series;
    arb_t t;

    arb_init(t);
    hypgeom_init(series);

    fmpz_poly_set_str(series->A, "6  273266 2836265 11114067 20691522 18408276 6291432");
    fmpz_poly_set_str(series->B, "1  1");
    fmpz_poly_set_str(series->P, "7  0 -10 137 -675 1530 -1620 648");
    fmpz_poly_set_str(series->Q, "7  591360 13328384 99442688 342097920 595722240 509607936 169869312");

    prec += FLINT_CLOG2(prec);
    arb_hypgeom_infsum(s, t, series, prec, prec);
    arb_mul_ui(t, t, 394240, prec);
    arb_div(s, s, t, prec);

    hypgeom_clear(series);
    arb_clear(t);
}
```

is faster than the current implementation. However, the 4-term Machin formula is even more efficient on a single core, and gives log(3), log(5) and log(7) at the same time. So possibly whenever we compute and cache log(2), we should compute and cache the others at the same time.

**Comment by fredrik-johansson** [comment]
Jorge Zuniga pointed me to this paper by Guillera https://anamat.unizar.es/jguillera/other/Guillera-ramalogs.pdf featuring Ramanujan-like series for simultaneous computation of log(2), log(3), log(5), ... which should outperform the Machin formulas.

> log(2), log(3) and log(5)  are obtained from Eq.(6), Eq.(11) and Eq.(16),
> log(7) is obtained adding Eq.(5) and
> log(11) is got adding Eq.(7)

**Comment by fredrik-johansson** [comment]
An even better log(2) formula is now available in #1767.
